cypress/e2e/tests/pages/explorer/dashboard/websockets/connection.spec.ts

```diff
@@ -0,0 +1,88 @@
+function executeWebSocket(command: string, podName: string, namespace: string, containerName: string, bearerToken:string) {
+  const websocketUrl = Cypress.env('api').replace('https', 'wss');
+
+  return cy.setupWebSocket(
+    websocketUrl,
+    namespace,
+    podName,
+    containerName,
+    command,
+    bearerToken
+  );
+}
+
+describe('Pod management and WebSocket interaction', { tags: ['@adminUser', '@standardUser', '@explorer'] }, () => {
+  let token:string;
+  const tokenDesc = 'e2e-test-description';
+  let tokenId:string;
+
+  before(() => {
+    cy.login();
+    // create tokens
+    cy.createToken(tokenDesc, 3600000, false).then((resp: Cypress.Response<any>) => {
+      token = resp?.body?.token;
+      tokenId = resp?.body?.id;
+    });
+  });
+  let podId = '';
+  let podName = `e2e-test`;
+  const projName = `project${ +new Date() }`;
+  const nsName = `namespace${ +new Date() }`;
+
+  it('should create a new pod', () => {
+    // get user id
+    cy.getRancherResource('v3', 'users?me=true').then((resp: Cypress.Response<any>) => {
+      const userId = resp.body.data[0].id.trim();
+
+      // create project
+      cy.createProject(projName, 'local', userId).then((resp: Cypress.Response<any>) => {
+        cy.wrap(resp.body.id.trim()).as('projId');
+
+        // create ns
+        cy.get<string>('@projId').then((projId) => {
+          cy.createNamespaceInProject(nsName, projId);
+        });
+
+        // create pod
+        cy.createPod(nsName, podName, 'nginx:stable-alpine3.20-perl').then((resp) => {
+          podName = resp.body.metadata.name;
+          podId = resp.body.metadata.id;
+        });
+
+        // Wait until pod state is running
+        cy.waitForRancherResource('v1', 'pods', podId, (resp) => {
+          const podObject = resp.body.data.find((obj) => obj.id === `${ nsName }/${ podName }`);
+
+          return podObject ? podObject.status.phase === 'Running' : false;
+        }, 3).then((result) => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(result).to.be.true;
+        });
+      });
+    });
+  });
+
+  it('should create a new folder', () => {
+    executeWebSocket('mkdir test-directory && echo "Directory created successfully"', podName, nsName, 'container-0', token).then((messages) => {
+      expect(messages[2]).to.include('Directory created successfully');
+    });
+  });
+
+  it('should validate the folder name', () => {
```

Comment:
It seems that the `messages`  variable was undefined in the last CI run for `validate the folder name`. 
Same for `should delete the folder`.

```
AssertionError: object tested must be an array, a map, an object, a set, a string, or a weakset, but undefined given
```